Move environment variables from Vercel to code

biendeveloper opened the pull request:
## Dework Task

https://app.dework.xyz/nation3/app-2?taskId=307bc999-e381-4919-b569-f2a7465f3dbd

## Related GitHub Issue

Fixes https://github.com/nation3/app/issues/53

## Admin checklist:

- [x]  On vercel, rename `NEXT_PUBLIC_NFTSTORAGE_KEY` with `NFTSTORAGE_KEY`
- [ ]  Remove all `NEXT_PUBLIC_*` variables except for: 
    - `NEXT_PUBLIC_CHAIN`
    - `NEXT_PUBLIC_INFURA_ID`
    - `NEXT_PUBLIC_ALCHEMY_ID`
    - `NEXT_PUBLIC_ETHERSCAN_ID`
    - `NFTSTORAGE_KEY`

aahna-ashina commented:
Thank you, @biendeveloper 😃

I'm new to the `NEXT_PUBLIC_NFTSTORAGE_KEY` variable.  What is it for?

biendeveloper commented:
It is used in `app/ui/pages/api/store-signature.ts`, It is used to store signatures in this service https://nft.storage/ (signatures are being sent from `app/ui/lib/sign-agreement.ts`

aahna-ashina commented:
@biendeveloper I see, thank you.  So that variable has to be kept private, I presume?

biendeveloper commented:
> @biendeveloper I see, thank you. So that variable has to be kept private, I presume?

Good point @aahna-ashina :) I removed `NEXT_PUBLIC_` prefix. so the front end won't catch this env 
Ref: https://vercel.com/guides/how-to-add-vercel-environment-variables#accessing-environment-variables

aahna-ashina commented:
@xPi2 Is there a way to test the NFT storage functionality on Goerli?

aahna-ashina reviewed (changes requested):
@biendeveloper Looks like the integration tests are not passing:

> <img width="862" alt="Screen Shot 2022-09-06 at 6 44 53 PM" src="https://user-images.githubusercontent.com/95955389/188615933-5cab23ed-4535-496f-b857-52829515a3f8.png">

Do you have access to click "Details" and see the log output in GitHub Actions?  If not, you might be able to reproduce this by running `yarn e2e:headless` locally.

aahna-ashina commented:
> I removed `NEXT_PUBLIC_` prefix

@biendeveloper Note that if you _rename_ an existing Vercel variable, the code will stop working once we merge this PR into `main`.  So please leave a note in the PR description for Vercel Admins to know what to do if you will be needing their help.

xPi2 reviewed (changes requested):
To be consistent if we do this move then I think all variables related with the contracts should be fetched from the deployments manifest.

Also, I don't understand how going from 3 different RPC providers to one is a good move.

biendeveloper commented:
- Fixed test cases
- Updated description
- > To be consistent if we do this move then I think all variables related to the contracts should be fetched from the deployments manifest.

   @xPi2 We are already using variables from the deployments manifest, is there an example to show which variables can be fetched?
- > Also, I don't understand how going from 3 different RPC providers to one is a good move. 

  I don't feel good about maintaining these three keys, especially in UI code, I can see all your API keys (someone can send a lot of traffic to these and break things):
  
  ![image](https://user-images.githubusercontent.com/91308782/188648321-e740164b-6f9b-4c75-a399-9cfa98020a41.png)

  I think if you are looking for fallback providers, better to use a few other trusted public nodes, and use [FallbackProvider](https://docs.ethers.io/v5/api/providers/other/#FallbackProvider) instead

xPi2 commented:
>       @xPi2 We are already using variables from the deployments manifest, is there an example to show which variables can be fetched?

Adding those variables to the manifest, making them even easier to access. I've already pushed the suggestions.
  
>       I don't feel good about maintaining these three keys, especially in UI code, I can see all your API keys (someone can send a lot of traffic to these and break things):
>       ![image](https://user-images.githubusercontent.com/91308782/188648321-e740164b-6f9b-4c75-a399-9cfa98020a41.png)

The fact you can see them is the problem, this keys should be private and we should allways fallback to public nodes if this are not provided. I think that's the course of action we should take.

aahna-ashina commented on contracts/deployments/goerli.json:
```diff
@@ -7,5 +7,7 @@
   "nationPassportNFT": "0x51F728c58697aFf9582cFDe3cBD00EC83E9ae7FC",
-  "nationPassportNFTIssuer": "0x8c16926819AB30B8b29A8E23F5C230d164337093"
+  "nationPassportNFTIssuer": "0x8c16926819AB30B8b29A8E23F5C230d164337093",
+    "nationPassportAgreementStatement": "By claiming a Nation3 passport I agree to the terms defined in the following URL",
+    "nationPassportAgreementURI": "https://bafkreiadlf3apu3u7blxw7t2yxi7oyumeuzhoasq7gqmcbaaycq342xq74.ipfs.dweb.link"
```
```suggestion
  "nationPassportAgreementStatement": "By claiming a Nation3 passport I agree to the terms defined in the following URL",
  "nationPassportAgreementURI": "https://bafkreiadlf3apu3u7blxw7t2yxi7oyumeuzhoasq7gqmcbaaycq342xq74.ipfs.dweb.link"
```

aahna-ashina commented on ui/.env.goerli:
```diff
@@ -1,5 +1,2 @@
-NEXT_PUBLIC_VENATION_REQUIRED_STAKE=2
-NEXT_PUBLIC_VENATION_REWARDS_MULTIPLIER=2.5
\ No newline at end of file
+NFTSTORAGE_KEY=
```
@xPi2 Is there a way to test the NFT storage functionality on Goerli?

biendeveloper replied:
I think it is the same as mainnet, create an account here https://nft.storage/  and add an API key, and once we sign a message on the join page, check if that signature is stored in https://nft.storage/files/

aahna-ashina replied:
@biendeveloper Okay, I created a test API key and added it to `.eng.goerli`

biendeveloper replied:
Yep correct :)

aahna-ashina commented on ui/pages/api/store-signature.ts:
```diff
@@ -6,7 +6,7 @@ import { provider } from '../../lib/connectors'
 
 const client = new NFTStorage({
-  token: process.env.NEXT_PUBLIC_NFTSTORAGE_KEY || '',
+  token: process.env.NFTSTORAGE_KEY || '',
```
@biendeveloper Would it be better to fetch this variable from `config.ts`?

biendeveloper replied:
I think it is not required to move to config because the config file is accessible by the front end, better to leave it this way

aahna-ashina replied:
@biendeveloper Oh, okay, good point.  Thanks for clarifying 😄

aahna-ashina commented on ui/.env.goerli:
```diff
@@ -1,5 +1,2 @@
-NEXT_PUBLIC_VENATION_REQUIRED_STAKE=2
-NEXT_PUBLIC_VENATION_REWARDS_MULTIPLIER=2.5
\ No newline at end of file
+NFTSTORAGE_KEY=
```
```suggestion
NFTSTORAGE_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJkaWQ6ZXRocjoweDRlOTU2NDk3NjI4ZTdGMEQyODQxZWZkNEIwQ
```

aahna-ashina commented:
@luisivan Could you handle the Vercel admin tasks?
